src/reactionsystem_conversions.jl

```diff
@@ -554,19 +554,41 @@ function Base.convert(::Type{<:NonlinearSystem}, rs::ReactionSystem; name = name
         as_odes = false, include_zero_odes)
     eqs, us, ps, obs, defs = addconstraints!(eqs, fullrs, ists, ispcs; remove_conserved)
 
+    # Makes modifications to handle conservation laws. Currently requires special consideration 
+    # due to MTK updates. Long-term, hopefully we can have a unified solution across all systems.
+    # This, current, NonlinearSystem solution is based on https://github.com/SciML/ModelingToolkit.jl/issues/3458#issuecomment-2725487268.
+    eqs, obs, initialization_eqs = handle_nlsys_cons_laws(rs, eqs, obs, remove_conserved)
+    us = union(species(rs), us)
```

Comment:
Essentially, `addconstraints` removes the elimiantesspecies from `us`. But using the new MTK approach these needs to be there, so I add them back in. I think this should work, but you might know some obscure side case where it does not (alternatively we can loop through the conservation equations and fetch them from there).

Reply:
I think a better approach would be to have a kwarg to addconstraints that just does the right changes for NonlinearSystems (or a version of it for nonlinear systems), rather than trying to remove and move around stuff.

Also, at this point in the conversion function you should be using `fullrs`, the flattened system, not `rs`.

Reply:
We can make `addconstraints` return the initialization equations as the final option (which can just be empty if there are none as for other systems). Then we pass them on. 

Another advantage of modifying it to do things the correct way would be that it is subsequently trivial to swap to it for other systems types later on.

Reply:
Basically, just add a kwarg like `treat_obs_as_eqs`, and have a sub-case for whether it is true or false (the default) in this code: https://github.com/SciML/Catalyst.jl/blob/06aed2e7d13ff79d2895e61651f4ca8a778e5868/src/reactionsystem_conversions.jl#L391-L401